source/frontend/src/features/contacts/contact/create/components/address/Address.tsx

```diff
@@ -51,20 +57,20 @@ export const Address: React.FunctionComponent<React.PropsWithChildren<IAddressPr
 
   // this counter determines how many address lines we render in the form; e.g. street1, street2, etc
   const { count, increment, decrement } = useCounter({
-    initial: addressLines ?? 1,
+    initial: addressLines || 1,
     min: 1,
     max: 3,
   });
-
   // clear extra address fields when they get removed from address form...
-  useEffect(() => {
```

Comment:
this useEffect used to have a race condition where sometimes the streetAddress3 and 2 were cleared during the load. updated this to a function instead to only trigger removal in response to a click.

Reply:
yeah this looks quite complex for a list of addresses